### ISSUE TITLE
Unexpected filament change ( 3.12.2)

### ISSUE THREAD

**Issue by domverity** [prusa3d/Prusa-Firmware#4061]
MK3S, just updated to 3.12.2, running the attached GCODE via serial from Octoprint, paused and behaved as if the GCODE instructed a filament change in the following approximate position (shown in Prusaslicer screenshot) during the first layer. I've never seen this behaviour before... I unloaded and reloaded the same filament and it continued as expected.

<img width="1081" alt="image" src="https://user-images.githubusercontent.com/11877701/222753393-260f4b88-e14a-4830-96e3-ce6bdc4b17f8.png">
 
[Merged_0.6n_0.3mm_PLA_MK3S_3h57m.gcode.zip](https://github.com/prusa3d/Prusa-Firmware/files/10883138/Merged_0.6n_0.3mm_PLA_MK3S_3h57m.gcode.zip)

**Comment by mattfox27** [comment]
Yep I'm getting the same bug as well, when using mmu about half of the filament changes pause and beep like m600 has been issued

**Comment by 3d-gussner** [comment]
@domverity @mattfox27 Thanks for reporting this issue.
@mattfox27 As you mention the MMU please describe it a bit more.

**Comment by mattfox27** [comment]
When doing multi material prints with MMU2s running 1.06fw 3.12.2 prusa firmware, sliced using latest prusaslicer.  About every hour or so into a 8hr multi material print it will stop and issue a manual firmware switch where it beeps and the screen will say filament changed correctly?  Yes/no.  Upon pressing yes mmu switches to correct filament and print resumes

**Comment by 3d-gussner** [comment]
@domverity Do you have by change the serial logging enabled in Octoprint? If so please share your log file to hopefully see more.
If not, please enable in OctoPrint the serial logging see https://help.prusa3d.com/article/crash-dump_364959#octoprint

Also please fill in the issue template when opening an issue, this helps a lot.


Printer type - [e.g. MK3S, MK3, MK2.5S, MK2.5, MK2S, MK2]
Printer firmware version - [e.g. 3.8.1, 3.8.1-RC1, ...]

MMU upgrade - [e.g. MMU2S, MMU2, MMU1]
MMU upgrade firmware version - [e.g. 1.0.6, 1.0.6-RC2, ...]

SD card or USB/Octoprint
Please let us know if you print via SD card or USB/Octoprint

Describe the bug
A clear and concise description of what the bug is.

To Reproduce
Please describe steps to reproduce the behavior.

Expected behavior
A clear and concise description of what you expected to happen.

G-code
Please attach a G-code. This will make it easier for us to replicate the error.

Video
Please attach a video. It usually helps to solve the problem.

**Comment by mattfox27** [comment]
I just had this happen again, im not using octoprint is there any logging being done with just SD Card?  I suppose I can hookup octoprint to capture logs if needed

**Comment by 3d-gussner** [comment]
@mattfox27 No we don't have the "power" and resources to log to the SD card. Please try to log with a serial terminal like Octoprint or putty.

**Comment by TwadaCZ** [comment]
Hi @3d-gussner ,
I report to Prusa helpdesk this issue from 3.12.1... I see this issue with 3.12.2 too... on my MK3s+ without MMU2...

Only if I print PETG, printer beep and want to uload and load filament... Randomly... Sometimes twice during 2 hours print... sometimes only during first 15 minutes... sometimes during last 15 minutes...

If I print PLA, there I did not see this issue with reload filament... during 20 hrs. prints...

I did not see it at my MK3s+ with MMU2, because there I did not print PETG...

When I downgrade to 3.11.0, everything working properly.

Thx, TwadaCZ

**Comment by domverity** [comment]
@3d-gussner Apologies for not following the template, I will do so in future.

I re-ran the gcode where I saw the issue with serial logging enabled in Octoprint but unfortunately the issue did not happen. I will continue printing with serial logging on and post a log if I catch it happening.

**Comment by mattfox27** [comment]
I was running octoprint with serial logging enabled and I caught it at about 21:37 is when it happened

> 2023-03-09 21:35:26,147 - Recv: T:205.1 /205.0 B:60.0 /60.0 T0:205.1 /205.0 @:63 B@:35 P:0.0 A:42.0
2023-03-09 21:35:28,043 - Recv: ok
2023-03-09 21:35:28,580 - Send: N333335 T1*13
2023-03-09 21:35:28,585 - Recv: T:205.1 /205.0 B:60.0 /60.0 T0:205.1 /205.0 @:62 B@:32 P:0.0 A:42.0
2023-03-09 21:35:28,683 - Recv: mmu_get_response - begin move: T-code
2023-03-09 21:35:28,711 - Recv: MMU <= 'T1'
2023-03-09 21:35:28,734 - Recv: Unloading finished 2
2023-03-09 21:35:30,136 - Recv: T:205.0 /205.0 B:60.0 /60.0 T0:205.0 /205.0 @:64 B@:31 P:0.0 A:42.0
2023-03-09 21:35:30,763 - Recv: mmu_get_response - begin move: load
2023-03-09 21:35:32,144 - Recv: T:205.0 /205.0 B:60.0 /60.0 T0:205.0 /205.0 @:65 B@:31 P:0.0 A:41.9
2023-03-09 21:35:34,151 - Recv: T:205.0 /205.0 B:60.0 /60.0 T0:205.0 /205.0 @:64 B@:24 P:0.0 A:41.9
2023-03-09 21:35:36,162 - Recv: T:205.1 /205.0 B:60.0 /60.0 T0:205.1 /205.0 @:62 B@:22 P:0.0 A:42.0
2023-03-09 21:35:37,600 - Recv: echo:busy: processing
2023-03-09 21:35:38,164 - Recv: T:205.1 /205.0 B:60.1 /60.0 T0:205.1 /205.0 @:63 B@:12 P:0.0 A:42.1
2023-03-09 21:35:40,168 - Recv: T:205.2 /205.0 B:60.1 /60.0 T0:205.2 /205.0 @:62 B@:23 P:0.0 A:41.9
2023-03-09 21:35:42,175 - Recv: T:205.0 /205.0 B:60.1 /60.0 T0:205.0 /205.0 @:65 B@:13 P:0.0 A:42.0
2023-03-09 21:35:44,178 - Recv: T:205.0 /205.0 B:60.1 /60.0 T0:205.0 /205.0 @:65 B@:19 P:0.0 A:41.8
2023-03-09 21:35:46,186 - Recv: T:205.2 /205.0 B:60.1 /60.0 T0:205.2 /205.0 @:61 B@:19 P:0.0 A:41.8
2023-03-09 21:35:46,603 - Recv: echo:busy: processing
2023-03-09 21:35:48,193 - Recv: T:205.2 /205.0 B:60.1 /60.0 T0:205.2 /205.0 @:62 B@:22 P:0.0 A:41.8
2023-03-09 21:35:49,204 - Recv: MMU <= 'A'
2023-03-09 21:35:49,798 - Recv: MMU => 'ok'
2023-03-09 21:35:49,851 - Recv: mmu_get_response() returning: 1
2023-03-09 21:35:50,197 - Recv: T:205.0 /205.0 B:60.0 /60.0 T0:205.0 /205.0 @:65 B@:32 P:0.0 A:41.9
2023-03-09 21:35:52,202 - Recv: T:205.1 /205.0 B:60.0 /60.0 T0:205.1 /205.0 @:62 B@:30 P:0.0 A:41.8
2023-03-09 21:35:54,206 - Recv: T:205.0 /205.0 B:60.0 /60.0 T0:205.0 /205.0 @:65 B@:31 P:0.0 A:41.9
2023-03-09 21:35:55,602 - Recv: echo:busy: processing
2023-03-09 21:35:55,767 - Recv: MMU can_load:
2023-03-09 21:35:56,214 - Recv: OOOOOOOOOOOOOOOOOOOOOOT:205.2 /205.0 B:60.1 /60.0 T0:205.2 /205.0 @:62 B@:19 P:0.0 A:41.9
2023-03-09 21:35:56,351 - Recv: OOOOOOOO succeeded.
2023-03-09 21:35:56,367 - Recv: ok
2023-03-09 21:35:56,398 - Send: N333336 G1 F10800*66
2023-03-09 21:35:56,430 - Recv: ok
2023-03-09 21:35:56,462 - Send: N333337 G1  X184.781 Y132.975*53
2023-03-09 21:35:56,481 - Recv: ok
2023-03-09 21:35:56,522 - Send: N333338 M900 K0.05*97
2023-03-09 21:35:56,540 - Recv: LA10C: Linear Advance mode: 1.5
2023-03-09 21:35:56,548 - Recv: echo:Advance K=0.05
2023-03-09 21:35:56,557 - Recv: ok
2023-03-09 21:35:56,573 - Send: N333339 M900 K30*120
2023-03-09 21:35:56,589 - Recv: K out of allowed range!
2023-03-09 21:35:56,599 - Recv: echo:Advance K=0.05
2023-03-09 21:35:56,607 - Recv: ok
2023-03-09 21:35:56,631 - Send: N333340 G4 S0*90
2023-03-09 21:35:56,655 - Recv: ok
2023-03-09 21:35:56,708 - Send: N333341 G1 E12.0000 F1140*55
2023-03-09 21:35:56,731 - Recv: ok
2023-03-09 21:35:56,756 - Send: N333342 G1  X235.336   E42.0000 F1011*64
2023-03-09 21:35:56,786 - Recv: ok
2023-03-09 21:35:56,803 - Send: N333343 G1  X184.781   E6.0000 F708*78
2023-03-09 21:35:56,826 - Recv: ok
2023-03-09 21:35:56,843 - Send: N333344 G1  Y132.475*105
2023-03-09 21:35:56,888 - Recv: ok
2023-03-09 21:35:56,906 - Send: N333345 G1  X176.961  E0.2972 F1584*90
2023-03-09 21:35:56,926 - Recv: ok
2023-03-09 21:35:56,943 - Send: N333346 G1  X176.211 Y132.975   E0.0343*77
2023-03-09 21:35:56,962 - Recv: ok
2023-03-09 21:35:56,977 - Send: N333347 G1  X234.961   E2.2329 F1800*122
2023-03-09 21:35:57,005 - Recv: ok
2023-03-09 21:35:57,020 - Send: N333348 G1  X235.711 Y133.475   E0.0343*78
2023-03-09 21:35:57,040 - Recv: ok
2023-03-09 21:35:57,055 - Send: N333349 G1  X176.961   E2.2329 F2198*122
2023-03-09 21:35:57,080 - Recv: ok
2023-03-09 21:35:57,096 - Send: N333350 G1  Y133.975  E0.0190*51
2023-03-09 21:35:57,118 - Recv: ok
2023-03-09 21:35:57,133 - Send: N333351 G1  X234.961   E2.2044 F4200*122
2023-03-09 21:35:57,335 - Recv: ok
2023-03-09 21:35:57,352 - Send: N333352 G1  Y134.475  E0.0190*59
2023-03-09 21:35:58,220 - Recv: T:205.2 /205.0 B:59.9 /60.0 T0:205.2 /205.0 @:61 B@:45 P:0.0 A:41.9
2023-03-09 21:35:58,871 - Recv: ok
2023-03-09 21:35:58,895 - Send: N333353 G1  X176.961   E2.2044 F4250*120
2023-03-09 21:36:00,227 - Recv: T:205.3 /205.0 B:60.0 /60.0 T0:205.3 /205.0 @:60 B@:36 P:0.0 A:42.0
2023-03-09 21:36:02,230 - Recv: T:205.0 /205.0 B:60.0 /60.0 T0:205.0 /205.0 @:64 B@:22 P:0.0 A:41.9
2023-03-09 21:36:02,545 - Recv: ok
2023-03-09 21:36:02,561 - Send: N333354 G1  X176.211 Y134.975   E0.0343*72
2023-03-09 21:36:04,233 - Recv: T:205.0 /205.0 B:60.0 /60.0 T0:205.0 /205.0 @:63 B@:19 P:0.0 A:42.0
2023-03-09 21:36:04,688 - Recv: ok
2023-03-09 21:36:04,704 - Send: N333355 G1  X234.961   E2.2329 F4300*119
2023-03-09 21:36:05,037 - Recv: ok
2023-03-09 21:36:05,054 - Send: N333356 G1  X235.711 Y135.475   E0.0343*71
2023-03-09 21:36:05,076 - Recv: ok
2023-03-09 21:36:05,103 - Send: N333357 G1  X176.961   E2.2329 F4350*117
2023-03-09 21:36:06,240 - Recv: T:204.8 /205.0 B:60.1 /60.0 T0:204.8 /205.0 @:67 B@:17 P:0.0 A:41.9
2023-03-09 21:36:07,055 - Recv: ok
2023-03-09 21:36:07,071 - Send: N333358 G1  Y135.975  E0.0190*61
2023-03-09 21:36:07,096 - Recv: ok
2023-03-09 21:36:07,112 - Send: N333359 G1  X234.961   E2.2044 F4400*116
2023-03-09 21:36:08,259 - Recv: T:205.1 /205.0 B:60.0 /60.0 T0:205.1 /205.0 @:62 B@:21 P:0.0 A:41.8
2023-03-09 21:36:08,501 - Recv: fsensor_checkpoint_print
2023-03-09 21:36:08,523 - Recv: ok
2023-03-09 21:36:08,543 - Send: N333360 G1  Y136.475  E0.0190*56
2023-03-09 21:36:08,548 - Recv: Resend: 333349
2023-03-09 21:36:08,591 - Recv: ok
2023-03-09 21:36:08,622 - Send: N333349 G1  X176.961   E2.2329 F2198*122
2023-03-09 21:36:08,632 - Recv: LCD status changed
2023-03-09 21:36:08,646 - Recv: echo:Enqueing to the front: "M600"
2023-03-09 21:36:08,656 - Recv: Error:Line Number is not Last Line Number+1, Last Line: 333348
2023-03-09 21:36:08,668 - Recv: Resend: 333349
2023-03-09 21:36:08,698 - Recv: ok
2023-03-09 21:36:08,716 - Send: N333350 G1  Y133.975  E0.0190*51
2023-03-09 21:36:10,267 - Recv: T:204.9 /205.0 B:60.1 /60.0 T0:204.9 /205.0 @:65 B@:13 P:0.0 A:42.0
2023-03-09 21:36:12,281 - Recv: T:204.6 /205.0 B:60.1 /60.0 T0:204.6 /205.0 @:70 B@:22 P:0.0 A:41.9
2023-03-09 21:36:14,281 - Recv: T:204.9 /205.0 B:60.1 /60.0 T0:204.9 /205.0 @:64 B@:23 P:0.0 A:41.9
2023-03-09 21:36:16,284 - Recv: T:204.9 /205.0 B:60.1 /60.0 T0:204.9 /205.0 @:65 B@:17 P:0.0 A:41.9
2023-03-09 21:36:17,509 - Recv: echo:busy: processing
2023-03-09 21:36:18,292 - Recv: T:205.3 /205.0 B:60.1 /60.0 T0:205.3 /205.0 @:58 B@:8 P:0.0 A:41.9
2023-03-09 21:36:18,520 - Recv: mmu_get_response - begin move: unload
2023-03-09 21:36:18,528 - Recv: MMU <= 'U0'
2023-03-09 21:36:18,644 - Recv: Unloading finished 1
2023-03-09 21:36:20,330 - Recv: T:205.9 /205.0 B:60.1 /60.0 T0:205.9 /205.0 @:49 B@:12 P:0.0 A:41.8
2023-03-09 21:36:22,363 - Recv: T:206.2 /205.0 B:60.1 /60.0 T0:206.2 /205.0 @:46 B@:8 P:0.0 A:41.7
2023-03-09 21:36:24,425 - Recv: T:206.1 /205.0 B:60.1 /60.0 T0:206.1 /205.0 @:49 B@:21 P:0.0 A:41.8
2023-03-09 21:36:26,403 - Recv: T:206.2 /205.0 B:60.1 /60.0 T0:206.2 /205.0 @:48 B@:17 P:0.0 A:41.8
2023-03-09 21:36:26,529 - Recv: echo:busy: processing
2023-03-09 21:36:28,404 - Recv: MMU => 'ok'
2023-03-09 21:36:28,431 - Recv: T:206.1 /205.0 B:60.1 /60.0 T0:206.1 /205.0 @:50 B@:16 P:0.0 A:41.7
2023-03-09 21:36:28,487 - Recv: mmu_get_response() returning: 1
2023-03-09 21:36:30,416 - Recv: T:205.8 /205.0 B:60.1 /60.0 T0:205.8 /205.0 @:54 B@:21 P:0.0 A:41.7
2023-03-09 21:36:32,423 - Recv: T:205.5 /205.0 B:60.0 /60.0 T0:205.5 /205.0 @:59 B@:29 P:0.0 A:41.7
2023-03-09 21:36:34,434 - Recv: T:205.4 /205.0 B:60.0 /60.0 T0:205.4 /205.0 @:59 B@:24 P:0.0 A:41.8
2023-03-09 21:36:35,531 - Recv: echo:busy: paused for user
2023-03-09 21:36:36,441 - Recv: T:205.3 /205.0 B:60.0 /60.0 T0:205.3 /205.0 @:60 B@:23 P:0.0 A:41.8
2023-03-09 21:36:38,448 - Recv: T:205.5 /205.0 B:60.0 /60.0 T0:205.5 /205.0 @:55 B@:23 P:0.0 A:41.7
2023-03-09 21:36:40,457 - Recv: T:205.8 /205.0 B:60.0 /60.0 T0:205.8 /205.0 @:49 B@:38 P:0.0 A:41.8
2023-03-09 21:36:42,466 - Recv: T:205.9 /205.0 B:60.0 /60.0 T0:205.9 /205.0 @:47 B@:31 P:0.0 A:41.7
2023-03-09 21:36:44,473 - Recv: T:205.9 /205.0 B:60.1 /60.0 T0:205.9 /205.0 @:47 B@:14 P:0.0 A:41.7
2023-03-09 21:36:44,549 - Recv: echo:busy: paused for user
2023-03-09 21:36:46,481 - Recv: T:205.7 /205.0 B:60.0 /60.0 T0:205.7 /205.0 @:51 B@:25 P:0.0 A:41.7
2023-03-09 21:36:48,488 - Recv: T:205.5 /205.0 B:60.1 /60.0 T0:205.5 /205.0 @:54 B@:15 P:0.0 A:41.7
2023-03-09 21:36:50,496 - Recv: T:205.3 /205.0 B:60.0 /60.0 T0:205.3 /205.0 @:56 B@:38 P:0.0 A:41.6
2023-03-09 21:36:52,504 - Recv: T:205.0 /205.0 B:59.9 /60.0 T0:205.0 /205.0 @:60 B@:44 P:0.0 A:41.6
2023-03-09 21:36:53,534 - Recv: echo:busy: paused for user
2023-03-09 21:36:54,514 - Recv: T:205.1 /205.0 B:60.0 /60.0 T0:205.1 /205.0 @:56 B@:21 P:0.0 A:41.6
2023-03-09 21:36:56,520 - Recv: T:204.9 /205.0 B:60.0 /60.0 T0:204.9 /205.0 @:59 B@:17 P:0.0 A:41.7
2023-03-09 21:36:58,528 - Recv: T:205.2 /205.0 B:60.0 /60.0 T0:205.2 /205.0 @:53 B@:27 P:0.0 A:41.5
2023-03-09 21:37:00,534 - Recv: T:205.4 /205.0 B:60.0 /60.0 T0:205.4 /205.0 @:51 B@:11 P:0.0 A:41.6
2023-03-09 21:37:02,534 - Recv: echo:busy: paused for user
2023-03-09 21:37:02,559 - Recv: T:205.3 /205.0 B:60.0 /60.0 T0:205.3 /205.0 @:52 B@:14 P:0.0 A:41.7
2023-03-09 21:37:04,549 - Recv: T:205.4 /205.0 B:60.0 /60.0 T0:205.4 /205.0 @:50 B@:15 P:0.0 A:41.6
2023-03-09 21:37:06,552 - Recv: T:205.4 /205.0 B:60.0 /60.0 T0:205.4 /205.0 @:51 B@:16 P:0.0 A:41.6
2023-03-09 21:37:08,560 - Recv: T:205.3 /205.0 B:60.0 /60.0 T0:205.3 /205.0 @:52 B@:28 P:0.0 A:41.6
2023-03-09 21:37:10,567 - Recv: T:205.0 /205.0 B:60.1 /60.0 T0:205.0 /205.0 @:57 B@:13 P:0.0 A:41.6
2023-03-09 21:37:11,537 - Recv: echo:busy: paused for user
2023-03-09 21:37:12,573 - Recv: T:204.6 /205.0 B:60.1 /60.0 T0:204.6 /205.0 @:61 B@:14 P:0.0 A:41.5
2023-03-09 21:37:14,576 - Recv: T:204.9 /205.0 B:60.0 /60.0 T0:204.9 /205.0 @:57 B@:27 P:0.0 A:41.6
2023-03-09 21:37:15,065 - Recv: mmu_get_response - begin move: load
2023-03-09 21:37:15,074 - Recv: MMU <= 'T1'
2023-03-09 21:37:16,584 - Recv: T:204.8 /205.0 B:60.0 /60.0 T0:204.8 /205.0 @:58 B@:25 P:0.0 A:41.6
2023-03-09 21:37:18,629 - Recv: T:205.1 /205.0 B:60.0 /60.0 T0:205.1 /205.0 @:52 B@:29 P:0.0 A:41.6
2023-03-09 21:37:20,532 - Recv: echo:busy: processing
2023-03-09 21:37:20,593 - Recv: T:205.1 /205.0 B:59.9 /60.0 T0:205.1 /205.0 @:52 B@:35 P:0.0 A:41.5
2023-03-09 21:37:22,600 - Recv: T:205.1 /205.0 B:59.9 /60.0 T0:205.1 /205.0 @:53 B@:41 P:0.0 A:41.5
2023-03-09 21:37:24,551 - Recv: MMU <= 'A'
2023-03-09 21:37:24,607 - Recv: T:205.1 /205.0 B:59.9 /60.0 T0:205.1 /205.0 @:52 B@:32 P:0.0 A:41.5
2023-03-09 21:37:25,149 - Recv: MMU => 'ok'
2023-03-09 21:37:25,159 - Recv: mmu_get_response() returning: 1
2023-03-09 21:37:26,612 - Recv: T:205.1 /205.0 B:60.0 /60.0 T0:205.1 /205.0 @:53 B@:27 P:0.0 A:41.6
2023-03-09 21:37:28,619 - Recv: T:205.0 /205.0 B:60.0 /60.0 T0:205.0 /205.0 @:54 B@:13 P:0.0 A:41.6
2023-03-09 21:37:29,535 - Recv: echo:busy: processing
2023-03-09 21:37:30,620 - Recv: T:204.9 /205.0 B:60.1 /60.0 T0:204.9 /205.0 @:57 B@:7 P:0.0 A:41.4
2023-03-09 21:37:31,063 - Recv: MMU can_load:
2023-03-09 21:37:31,637 - Recv: OOOOOOOOOOOOOOOOOOOOOOOOOOOOOO succeeded.
2023-03-09 21:37:32,628 - Recv: T:204.7 /205.0 B:60.0 /60.0 T0:204.7 /205.0 @:59 B@:19 P:0.0 A:41.4
2023-03-09 21:37:34,635 - Recv: T:204.8 /205.0 B:60.1 /60.0 T0:204.8 /205.0 @:56 B@:18 P:0.0 A:41.5
2023-03-09 21:37:36,640 - Recv: T:204.8 /205.0 B:60.1 /60.0 T0:204.8 /205.0 @:56 B@:13 P:0.0 A:41.5
2023-03-09 21:37:38,536 - Recv: echo:busy: processing
2023-03-09 21:37:38,647 - Recv: T:204.9 /205.0 B:60.1 /60.0 T0:204.9 /205.0 @:55 B@:17 P:0.0 A:41.3
2023-03-09 21:37:40,651 - Recv: T:204.3 /205.0 B:60.0 /60.0 T0:204.3 /205.0 @:64 B@:19 P:0.0 A:41.4
2023-03-09 21:37:42,656 - Recv: T:204.2 /205.0 B:60.0 /60.0 T0:204.2 /205.0 @:66 B@:34 P:0.0 A:41.5
2023-03-09 21:37:44,659 - Recv: T:203.4 /205.0 B:60.0 /60.0 T0:203.4 /205.0 @:76 B@:28 P:0.0 A:41.3
2023-03-09 21:37:46,666 - Recv: T:203.7 /205.0 B:59.9 /60.0 T0:203.7 /205.0 @:71 B@:38 P:0.0 A:41.4
2023-03-09 21:37:47,534 - Recv: echo:busy: paused for user
2023-03-09 21:37:48,669 - Recv: T:203.8 /205.0 B:59.9 /60.0 T0:203.8 /205.0 @:69 B@:33 P:0.0 A:41.4
2023-03-09 21:37:50,676 - Recv: T:204.5 /205.0 B:59.9 /60.0 T0:204.5 /205.0 @:59 B@:31 P:0.0 A:41.4
2023-03-09 21:37:52,679 - Recv: T:205.1 /205.0 B:60.0 /60.0 T0:205.1 /205.0 @:50 B@:20 P:0.0 A:41.4
2023-03-09 21:37:54,686 - Recv: T:206.0 /205.0 B:60.0 /60.0 T0:206.0 /205.0 @:39 B@:27 P:0.0 A:41.4
2023-03-09 21:37:56,537 - Recv: echo:busy: paused for user
2023-03-09 21:37:56,689 - Recv: T:206.1 /205.0 B:59.9 /60.0 T0:206.1 /205.0 @:40 B@:30 P:0.0 A:41.3
2023-03-09 21:37:58,704 - Recv: T:206.2 /205.0 B:59.9 /60.0 T0:206.2 /205.0 @:41 B@:32 P:0.0 A:41.4
2023-03-09 21:38:00,713 - Recv: T:206.0 /205.0 B:60.0 /60.0 T0:206.0 /205.0 @:45 B@:26 P:0.0 A:41.5
2023-03-09 21:38:02,715 - Recv: T:205.4 /205.0 B:60.0 /60.0 T0:205.4 /205.0 @:54 B@:9 P:0.0 A:41.4
2023-03-09 21:38:04,716 - Recv: T:204.9 /205.0 B:60.0 /60.0 T0:204.9 /205.0 @:62 B@:12 P:0.0 A:41.4
2023-03-09 21:38:05,589 - Recv: echo:busy: paused for user
2023-03-09 21:38:06,723 - Recv: T:204.6 /205.0 B:60.0 /60.0 T0:204.6 /205.0 @:65 B@:24 P:0.0 A:41.4
2023-03-09 21:38:08,733 - Recv: T:204.7 /205.0 B:60.1 /60.0 T0:204.7 /205.0 @:60 B@:14 P:0.0 A:41.3
2023-03-09 21:38:10,738 - Recv: T:204.9 /205.0 B:60.1 /60.0 T0:204.9 /205.0 @:57 B@:13 P:0.0 A:41.3
2023-03-09 21:38:12,742 - Recv: T:205.0 /205.0 B:60.0 /60.0 T0:205.0 /205.0 @:55 B@:23 P:0.0 A:41.4
2023-03-09 21:38:14,541 - Recv: echo:busy: paused for user
2023-03-09 21:38:14,747 - Recv: T:205.3 /205.0 B:60.0 /60.0 T0:205.3 /205.0 @:51 B@:20 P:0.0 A:41.3
2023-03-09 21:38:16,740 - Recv: T:205.5 /205.0 B:60.0 /60.0 T0:205.5 /205.0 @:48 B@:32 P:0.0 A:41.2
2023-03-09 21:38:18,770 - Recv: T:205.8 /205.0 B:60.0 /60.0 T0:205.8 /205.0 @:44 B@:34 P:0.0 A:41.3
2023-03-09 21:38:20,758 - Recv: T:205.6 /205.0 B:59.9 /60.0 T0:205.6 /205.0 @:48 B@:49 P:0.0 A:41.2

**Comment by mattfox27** [comment]
> 2023-03-09 21:36:08,646 - Recv: echo:Enqueing to the front: "M600"
> 2023-03-09 21:36:08,656 - Recv: Error:Line Number is not Last Line Number+1, Last Line: 333348
> 2023-03-09 21:36:08,668 - Recv: Resend: 333349

I posted a log but this might be the error causing it???

**Comment by schiele** [comment]
I observe the same problem as described here: During a PETG print of a gcode file from SD card with 2 intentional M600 color change commands included the printer starts behaving as if seen a M600 command in various different places, even long before the first of those commands even appears in the file. If you inspect the gcode file you will find that the first M600 command appears on layer Z=1.8. Still, the printer starts initiating manual filament changes towards the end of the first layer Z=0.2 in many places, typically after one perimeter line was completed before starting the next one despite the gcode file not including any command I would associate with such behavior.

The printer I reproduced this: MK3S+ with MMU2S, though I disconnected the MMU2S from the power source and fed the filament directly into the extruder, such that from a firmware point of view this is a pure MK3S+. Latest firmware (3.12.2) is installed. Printer is running in Stealth Mode.

Printing done from SD card.

gcode file: [protractor_upper_part_depressed_0.2mm_PETG_MK3S_2h14m.gcode.txt](https://github.com/prusa3d/Prusa-Firmware/files/10973773/protractor_upper_part_depressed_0.2mm_PETG_MK3S_2h14m.gcode.txt)

Steps to reproduce: Print the attached gcode file from SD card.

Expected behavior: Printer should only initiate manual filament changes when it hits a M600 command.
Observed behavior: Printer initiates lots of manual filament changes for no obvious reason.

**Comment by gudnimg** [comment]
@mattfox27 sorry for the late response, If you see this in the printer's serial log `Enqueing to the front: "M600"` then that  confirms the issue is filament runout. The fsensor reports to the firmware it no longer detects filament so the printer will try to recover the print by injecting M600 into the front of the queue.

@schiele without a log I can only assume you're seeing filament runout too. It's the only way the firmware will inject M600 command. Please check the logs from the printer or check if the filament sensor or FINDA is no longer detecting filament.

Note that when using an MMU, the FINDA sensor can also cause filament runout events. If the FINDA LED is illuminating red that means it doesnt detect filament.

I have seen issues with FINDA before when the filament is bent in a certain way, should be easily fixed by slightly tuning the FINDA position.

**Comment by PyBonnetainNesterenko** [comment]
Hello,
No MMU here, but Octoprint.
Just upgraded from fw 3.10 to 3.12.2, first print since, and I already got 3 requests for a filament change (despite a full roll of filament). This makes unattended prints almost impossible. On exactly the same physical setup this problem did not occur with fw 3.10 (did not test 3.11).
If it's the FINDA triggering, it seems to be more sensitive than previously, and wrongly so.
_Small edit : After checking Octoprint's logs, I went from 3.11.0 (not 3.10 as written above) to 3.12.2, if this detail could be useful._

**Comment by PyBonnetainNesterenko** [comment]
I enabled the serial.log on OctoPrint, ran a print, and here are the lines surronding the M600.
```
2023-04-05 23:15:34,324 - Recv: T:215.2 /215.0 B:61.4 /60.0 T0:215.2 /215.0 @:31 B@:0 P:0.0 A:33.6
2023-04-05 23:15:35,298 - Recv: echo:busy: processing
2023-04-05 23:15:36,327 - Recv: T:215.1 /215.0 B:61.6 /60.0 T0:215.1 /215.0 @:34 B@:0 P:0.0 A:33.3
2023-04-05 23:15:37,302 - Recv: echo:busy: processing
2023-04-05 23:15:38,334 - Recv: T:215.2 /215.0 B:61.7 /60.0 T0:215.2 /215.0 @:32 B@:0 P:0.0 A:33.3
2023-04-05 23:15:39,300 - Recv: echo:busy: processing
2023-04-05 23:15:40,337 - Recv: T:215.2 /215.0 B:61.7 /60.0 T0:215.2 /215.0 @:31 B@:0 P:0.0 A:33.4
2023-04-05 23:15:41,300 - Recv: echo:busy: processing
2023-04-05 23:15:41,680 - Recv: LCD status changed
2023-04-05 23:15:41,682 - Recv: ok
2023-04-05 23:15:41,696 - Send: N21 G1 Z0.2 F720*30
2023-04-05 23:15:41,738 - Recv: ok
2023-04-05 23:15:41,751 - Send: N22 G1 Y-3 F1000*24
2023-04-05 23:15:41,758 - Recv: ok
2023-04-05 23:15:41,768 - Send: N23 G92 E0*118
2023-04-05 23:15:41,775 - Recv: ok
2023-04-05 23:15:41,788 - Send: N24 G1 X60 E9 F1000*91
2023-04-05 23:15:41,799 - Recv: ok
2023-04-05 23:15:41,809 - Recv: fsensor_checkpoint_print
2023-04-05 23:15:41,813 - Send: N25 G1 X100 E12.5 F1000*76
2023-04-05 23:15:41,817 - Recv: Resend: 22
2023-04-05 23:15:41,829 - Recv: ok
2023-04-05 23:15:41,831 - Recv: LCD status changed
2023-04-05 23:15:41,833 - Recv: fsensor_update - M600
2023-04-05 23:15:41,837 - Send: N22 G1 Y-3 F1000*24
2023-04-05 23:15:41,835 - Recv: echo:Enqueing to the front: "M600"
2023-04-05 23:15:41,841 - Recv: Error:Line Number is not Last Line Number+1, Last Line: 21
2023-04-05 23:15:41,845 - Recv: Resend: 22
2023-04-05 23:15:41,853 - Recv: ok
2023-04-05 23:15:41,857 - Send: N23 G92 E0*118
2023-04-05 23:15:42,343 - Recv: T:215.2 /215.0 B:61.9 /60.0 T0:215.2 /215.0 @:30 B@:0 P:0.0 A:33.4
2023-04-05 23:15:43,819 - Recv: echo:busy: processing
2023-04-05 23:15:44,347 - Recv: T:214.9 /215.0 B:61.9 /60.0 T0:214.9 /215.0 @:34 B@:0 P:0.0 A:33.2
2023-04-05 23:15:45,821 - Recv: echo:busy: processing
2023-04-05 23:15:46,354 - Recv: T:215.1 /215.0 B:61.8 /60.0 T0:215.1 /215.0 @:32 B@:0 P:0.0 A:33.3
2023-04-05 23:15:47,821 - Recv: echo:busy: processing
```
So there is indeed (as far as I understand it) a filament runout wrongly detected at 2023-04-05 23:15:41,833.

**Comment by cristian-rivera** [comment]
As @PyBonnetainNesterenko mentioned, this could be the same issue as found here: https://github.com/prusa3d/Prusa-Firmware/issues/3872

We've been informed that the temporary removal of runout-related code shouldn't lead to this problem, but it seems to be causing some difficulties. My issue has been open for two months, and I'm concerned that it may not have received the attention it needs. I understand the dev team might be busy, but addressing the original issue earlier might have been helpful, as more people are encountering it now.

We can assist the dev team by trying a few things to help narrow down the source of the issue. It appears that all instances of this issue involve printing over serial using OctoPrint. While OctoPrint doesn't seem to be the problem, as it doesn't occur on version 3.11, we can still test the following:

- Attempt to print using the SD card, preferably with the same files that initially caused the issue.
- Try printing through PrusaLink, again, using the same files that initially caused the issue.

Does this issue occur when printing from the SD card or via PrusaLink? I recently set up PrusaLink in my Kubernetes cluster and installed version 3.12.2 on my MK3S+, so I'll be testing this myself as well.

**Comment by leptun** [comment]
We are aware of this issue. The problem is that the issue doesn't make sense in the first place given what was changed in code, so nothing should have resulted in this bug.
There are no plans for releasing any other 3.12.3 firmware. The current development is happening on the MK3 branch, which will be the future 3.13.0-RC1 firmware. At the moment I can't give you an estimate of when that will be released. The thing about 3.13 is that for the new MMU implementation, we have completely rewritten the filament sensor code, so whatever bug was there before should be gone now. However, I haven't been able to ever replicate this bug in the first place, so I can only hope that the issue is gone now.

I'm not sure how useful further debugging in 3.12.2 would be since I don't think the findings would apply to the new filament sensor code. I think we'll have to put this problem on hold until we release the first version with the new filament sensor code. Then we can start debugging that more easily if the problem persists.

**Comment by Prusa-Support** [comment]
Again we can't exclude a temporary filament outage detection, due to various reasons - some of which are mentioned in https://github.com/prusa3d/Prusa-Firmware/issues/3872#issuecomment-1413930200.

In the case of MMU, there are two filament sensors that need to be calibrated to be sensitive to filament outage but forgiving for little filament inconsistencies (deformed/undersized filament).

These kinds of problems may not be firmware related and could persist on older and newer firmware versions.
We will keep investigating and collecting feedback even after the new firmware release.
Until then, please make sure that your filament is regular, and that the filament sensor triggering mechanism is clean and calibrated adequately. Proper troubleshooting may solve the problem.

Michele Moramarco
Prusa Research

**Comment by felias-fogg** [comment]
Just experienced the same problem after I upgraded to the new firmware when printing the upgrade parts for an MK3S+ with the standard Prusa PETG. And it happened a couple of times while I printed. 

Firmware: 3.12.2 (MK3)
Model: MK3
Filament: black PETG from PRUSA
G-Code file: Y-axis-LCD_MK3StoMK3S+_0.2mm_PETG_MK3_1h43m.gcode (from printables.com)
Printing from SD-card, no MMU

After downgrading to 3.11.0, the problem vanished. So, this definitely looks like a firmware problem to me.

**Comment by TwadaCZ** [comment]
Yes, I reported it to chat support immediately after the release 3.12.0... and here (was closed) with 3.12.2 release...

I have same problem on my MK3s (without MMU!!!)... only with PETG print...

I can reproduce this issue alway when print PETG with any 3.12 firmware... every print.

If I downgrade to 3.11.x firmware and print on same printer, same material, same GCODE file... this work perfect and problem does not appear.

**Comment by idisjcmschoot** [comment]
I have partly the same problem release 3.12.1, the out of filament sensor is triggered during a print, it continues printing without user action but with a Y axis layershift. While continuing printing there is are more out of filament messages but creating a Y-axis shift in the opposite direction ([Like this](https://forum.prusa3d.com/forum/original-prusa-i3-mk3s-mk3-how-do-i-print-this-printing-help/layer-shifting-after-filament-change-in-the-middle-of-the-print/)) 
I am getting this with PETG and ASA fillaments and using a Octopi.
I did the print 4 times and I got 4 different results also with different models it fails. When I print the same in PLA it does not seem to occur.

I am going to try release 3.11.0

**Comment by Prusa-Support** [comment]
@idisjcmschoot The layer shift after a pause may be caused by a loose belt, excess lubricant, or anything else preventing an accurate Y-axis parking. Thanks for testing and comparing different firmware versions.

.

If the filament's material is relevant, there is a chance that a slightly more flexible material is being compressed under a too-tight idler tension screw. For example, PLA is quite stiff so it will keep its shape or get damaged rather than squishing; PETG is slightly flexible instead.
This is mere speculation though and wouldn't explain why the problem only persists on FW 3.12.x.

We will investigate further. Thanks for your input and for keep on testing.

Michele Moramarco
Prusa Research

**Comment by Prusa-Support** [comment]
Would you guys mind if this issue is closed as duplicate of #3872?

This one wasn't closed because it got escalated quite quickly, in parallel with #3872. However neither of the issues show significant progress (just yet).
I suggest closing this one because opened at a later time, and I encourage all users here to join forces in the other issue.

Michele Moramarco
Prusa Research

**Comment by domverity** [comment]
@Prusa-Support fine by me...

**Comment by Prusa-Support** [comment]
Please continue reporting on issue https://github.com/prusa3d/Prusa-Firmware/issues/3872.
This one will be closed as a duplicate now. Thanks.

Michele Moramarco
Prusa Research
